src/main/java/com/powsybl/openloadflow/sensi/SensitivityFactorReaderAdapter.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.openloadflow.sensi;
+
+import com.powsybl.contingency.Contingency;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.sensitivity.SensitivityFactor;
+import com.powsybl.sensitivity.SensitivityFactorsProvider;
+import com.powsybl.sensitivity.factors.BranchFlowPerInjectionIncrease;
+import com.powsybl.sensitivity.factors.BranchFlowPerLinearGlsk;
+import com.powsybl.sensitivity.factors.BranchFlowPerPSTAngle;
+import com.powsybl.sensitivity.factors.BranchIntensityPerPSTAngle;
+import com.powsybl.sensitivity.factors.functions.BranchFlow;
+import com.powsybl.sensitivity.factors.functions.BranchIntensity;
+import com.powsybl.sensitivity.factors.variables.InjectionIncrease;
+import com.powsybl.sensitivity.factors.variables.LinearGlsk;
+import com.powsybl.sensitivity.factors.variables.PhaseTapChangerAngle;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public class SensitivityFactorReaderAdapter implements SensitivityFactorReader {
+
+    private final Network network;
+
+    private final SensitivityFactorsProvider sensitivityFactorsProvider;
+
+    private final List<Contingency> contingencies;
+
+    public SensitivityFactorReaderAdapter(Network network, SensitivityFactorsProvider sensitivityFactorsProvider,
+                                          List<Contingency> contingencies) {
+        this.network = Objects.requireNonNull(network);
+        this.sensitivityFactorsProvider = Objects.requireNonNull(sensitivityFactorsProvider);
+        this.contingencies = Objects.requireNonNull(contingencies);
+    }
+
+    @Override
+    public void read(Handler handler) {
+        for (SensitivityFactor factor : sensitivityFactorsProvider.getCommonFactors(network)) {
+            read(handler, factor);
+        }
+        // FIXME additional factors are not yet supported
+        if (!sensitivityFactorsProvider.getAdditionalFactors(network).isEmpty()) {
```

Comment:
How do you think additional factors should be managed ? Do we pass additional info inside the context, or do we need to change the handler's methods ?

Reply:
We can add contingency ID to handler